src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,611 @@
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.DotNet.MSIdentity.AuthenticationParameters;
+using Microsoft.DotNet.MSIdentity.Properties;
+using Microsoft.DotNet.MSIdentity.Tool;
+
+namespace Microsoft.DotNet.MSIdentity.CodeReaderWriter
+{
+    internal class ProjectModifier
+    {
+        private List<CodeModifierConfig> _codeModifierConfigs;
+        private readonly ProvisioningToolOptions _toolOptions;
+        private readonly ApplicationParameters _appParameters;
+        private readonly IConsoleLogger _consoleLogger;
+        public SyntaxTrivia SemiColonTrivia
+        {
+            get
+            {
+                return SyntaxFactory.Trivia(SyntaxFactory.SkippedTokensTrivia()
+                    .WithTokens(SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.SemicolonToken))));
+            }
+        }
+        public ProjectModifier(ApplicationParameters applicationParameters, ProvisioningToolOptions toolOptions, IConsoleLogger consoleLogger)
+        {
+            _toolOptions = toolOptions ?? throw new ArgumentNullException(nameof(toolOptions));
+            _appParameters = applicationParameters ?? throw new ArgumentNullException(nameof(applicationParameters));
+            _consoleLogger = consoleLogger ?? throw new ArgumentNullException(nameof(consoleLogger));
+            _codeModifierConfigs = new List<CodeModifierConfig>();
+        }
+
+        internal IDictionary<string, string>? VerfiyParameters(string[]? parametersToCheck, List<ParameterSyntax> foundParameters)
+        {
+            IDictionary<string, string> parametersWithNames = new Dictionary<string, string>();
+            if (foundParameters.Any())
+            {
+                var pars = foundParameters.ToList();
+                if (parametersToCheck != null)
+                {
+                    foreach (var parameter in parametersToCheck)
+                    {
+                        var verifiedParams = pars.Where(p => p.Type != null && p.Type.ToFullString().Trim().Equals(parameter.Trim(), StringComparison.OrdinalIgnoreCase));
+                        if (!verifiedParams.Any())
+                        {
+                            return null;
+                        }
+                        else
+                        {
+                            parametersWithNames.Add(parameter, verifiedParams.First().Identifier.ValueText);
+                        }
+                    }
+                }
+            }
+            
+            return parametersWithNames;
+        }
+
+        /// <summary>
+        /// Added "Microsoft identity platform" auth to base or empty C# .NET Core 3.1, .NET 5 and above projects.
+        /// Includes adding PackageReferences, modifying Startup.cs, and Layout.cshtml changes.
+        /// </summary>
+        /// <param name="projectType"></param>
+        /// <returns></returns>
+        public async Task AddAuth()
```

Comment:
>AddAuth [](http://example.com/codeflow?start=26&length=7)

I am lost ... :) kind of large method huh?

Reply:
refactored and added DocumentBuilder.cs